content/en/blog/_posts/2024-08-19-pod-failure-policy-for-jobs-goes-ga.md

```diff
@@ -0,0 +1,181 @@
+of rules.
+
+For each rule you define _match requirements_ based on one of the following properties:
+
+- Container exit codes: the `onExitCodes` property.
+- Pod conditions: the `onPodConditions` property.
+
+Additionally, for each rule, you specify one of the following actions to take
+when a Pod matches the rule:
+- `Ignore`: Do not count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+- `FailJob`: Fail the entire Job and terminate all running Pods.
+- `FailIndex`: Fail the index corresponding to the failed Pod.
+  This action is used together with the [Backoff limit per index](/docs/concepts/workloads/controllers/job/#backoff-limit-per-index) feature.
+- `Count`: Count the failure towards the `backoffLimit` or `backoffLimitPerIndex`.
+  This is the default behavior.
+
+When Pod failures occur in a running Job, Kubernetes matches the
+failed Pod status against the list of Pod failure policy rules, in the specified
+order, and takes the corresponding actions for the first matched rule.
+
+Note that when specifying the Pod failure policy, you must also set the Job's
+Pod template with `restartPolicy: Never`. This is important to prevent race
+conditions between the Kubelet and Job controller when counting Pod failures.
+
+### Kubernetes-initiated Pod disruptions
+
+To allow matching Pod failure policy rules against failures caused by
+disruptions initiated by Kubernetes, this feature introduces the `DisruptionTarget`
+Pod condition.
+
+This condition is added by Kubernetes when a Pod fails because of a retriable
+[disruption scenario](/docs/concepts/workloads/pods/disruptions/#pod-disruption-conditions).
+The `DisruptionTarget` condition contains one of the following reasons that corresponds to
+these disruption scenarios:
+
+- `PreemptionByKubeScheduler`: [Preemption](/docs/concepts/scheduling-eviction/pod-priority-preemption)
+   by `kube-scheduler` to accommodate a new Pod that has a higher priority.
+- `DeletionByTaintManager` - the Pod is due to be deleted by
+   `kube-controller-manager` due to a `NoExecute` [taint](/docs/concepts/scheduling-eviction/taint-and-toleration/)
+   that the Pod doesn't tolerate.
+- `EvictionByEvictionAPI` - the Pod is due to be deleted by an
+   [API-initiated eviction](/docs/concepts/scheduling-eviction/api-eviction/).
+- `DeletionByPodGC` - the Pod is bound to a node that no longer exists, and is due to
+   be deleted by [Pod garbage collection](/docs/concepts/workloads/pods/pod-lifecycle/#pod-garbage-collection).
+- `TerminationByKubelet` - the Pod was terminated by
+  [graceful node shutdown](/docs/concepts/cluster-administration/node-shutdown/#graceful-node-shutdown),
+  [node pressure eviction](/docs/concepts/scheduling-eviction/node-pressure-eviction/)
+  or preemption for [system critical pods](/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/).
+
+Note that while the `DisruptionTarget` condition can be leveraged by the Pod
```

Comment:
FYI: added this without an explicit ask, but I find it might be useful to know by users, and not immediately obvious. cc @atiratree @shannonxtreme

Reply:
sgtm, updated following the suggested edit.